packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncher.js

```diff
@@ -23,81 +17,33 @@ export const propTypes = {
   /** Callback called when application launcher toggle is clicked */
   onToggle: PropTypes.func,
   /** Indicates where menu will be alligned horizontally */
-  position: PropTypes.oneOf(Object.values(ApplicationLauncherPosition)),
+  position: PropTypes.oneOf(Object.values(DropdownPosition)),
   /** Adds accessible text to the button. Required for plain buttons */
   'aria-label': PropTypes.string
 };
 
 export const defaultProps = {
   className: '',
-  direction: ApplicationLauncherDirection.down,
+  direction: DropdownDirection.down,
   dropdownItems: [],
   isOpen: false,
   onSelect: Function.prototype,
   onToggle: Function.prototype,
-  position: ApplicationLauncherPosition.left,
+  position: DropdownPosition.left,
   'aria-label': 'Actions'
 };
 
 class ApplicationLauncher extends React.Component {
-  onEnter = () => {
-    this.openedOnEnter = true;
-  };
-
-  componentDidUpdate() {
-    if (!this.props.isOpen) this.openedOnEnter = false;
-  }
-
   render() {
-    const {
-      'aria-label': ariaLabel,
-      children,
-      dropdownItems,
-      className,
-      isOpen,
-      onSelect,
-      onToggle,
-      ...props
-    } = this.props;
-    return (
-      <GenerateId>
-        {randomId => (
-          <div
-            className={css(styles.appLauncher, isOpen && styles.modifiers.expanded, className)}
-            ref={ref => {
-              this.parentRef = ref;
-            }}
-          >
-            {Children.map(
-              <Toggle id={`pf-toggle-id-${randomId}`} aria-label={ariaLabel} onToggle={onToggle}>
-                <ThIcon />
-              </Toggle>,
-              oneToggle =>
-                cloneElement(oneToggle, {
-                  parentRef: this.parentRef,
-                  id: randomId,
-                  isOpen,
-                  isPlain: true,
-                  ariaHasPopup: true,
-                  onEnter: this.onEnter
-                })
-            )}
-            {isOpen && (
-              <DropdownContext.Provider value={event => onSelect && onSelect(event)}>
-                <ApplicationLauncherMenu
-                  isOpen={isOpen}
-                  position="left"
-                  aria-labelledby={ariaLabel}
-                  openedOnEnter={this.openedOnEnter}
-                >
-                  {dropdownItems}
-                </ApplicationLauncherMenu>
-              </DropdownContext.Provider>
-            )}
-          </div>
-        )}
-      </GenerateId>
-    );
+    return <Dropdown
+      { ...this.props }
+      toggle={
+        <DropdownToggle iconComponent={null} onToggle={this.props.onToggle}>
```

Comment:
Can you check if a11y is the same? For example, before aria-label was on the button, now it;s on the outer div. Also the button does not have aria-expanded set to true like before

Reply:
> Also the button does not have aria-expanded set to true like before

I can see that is has aria-expanded, strange.

Before the a11y was broken on menu because it had `aria-labelledby="Actions"` where `"Actions"` was not any ID in DOM, and yeah the aria label should be on the button, not on div, gotta fix that by passing `aria-label` to toggle.

Reply:
Sorry, I can also see aria-expanded now